### PR TITLE
[logs] user is aware of the transport used by looking at the agent status

### PR DESCRIPTION
**Pull request by blemale**
### What does this PR do?

When running the Agent status we should have the following:
```
==========
Logs Agent
==========
    Sending <COMPRESSION_MODE> logs in <PROTOCOL> to <ENDPOINT> on port <PORT>
    
    LogsProcessed: 310804
    LogsSent: 310788
```

With:
`<COMPRESSION_MODE>`: compressed or uncompressed
`<PROTOCOL>`: SSL encrypted TCP or HTTPS
`<ENDPOINT>`: endpoint
`<PORT>`: Port used

### Motivation

Better visibility on the logs configuration for the user

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/logs/status/builder.go [hunk 1]
```diff
@@ -51,6 +55,43 @@ func (b *Builder) getIsRunning() bool {
...
+	return result
+}
+
+func (b *Builder) formatEndpoint(endpoint config.Endpoint, prefix string) string {
+	compression := "uncompressed"
+	if endpoint.UseCompression {
+		compression = "compressed"
+	}
+
+	protocol := ""
```
Comment (ajacquemot) [inline pkg/logs/status/builder.go]: nits 
```suggestion
	var protocol string
```

File: pkg/logs/status/builder.go [hunk 2]
```diff
@@ -51,6 +55,43 @@ func (b *Builder) getIsRunning() bool {
...
+	} else {
+		if endpoint.UseSSL {
+			protocol = "SSL encrypted TCP"
+		} else {
+			protocol = "TCP"
+		}
+	}
+
+	host := endpoint.Host
+
```
Comment (ajacquemot) [inline pkg/logs/status/builder.go]: nits: empty space, go code is meant to be compact

File: pkg/status/dist/templates/logsagent.tmpl [hunk 3]
```diff
@@ -10,6 +10,13 @@ Logs Agent
   Logs Agent is not running
 {{- end }}
 
+{{- if .endpoints }}
+
+  {{- range $endpoint := .endpoints }}
```
Comment (ajacquemot) [inline pkg/status/dist/templates/logsagent.tmpl]: Would probably nice to add a header for this section

File: pkg/logs/status/builder.go [hunk 4]
```diff
@@ -51,6 +55,43 @@ func (b *Builder) getIsRunning() bool {
...
+func (b *Builder) getEndpoints() []string {
+	result := make([]string, 0)
+	result = append(result, b.formatEndpoint(b.endpoints.Main, ""))
+	for _, additional := range b.endpoints.Additionals {
+		result = append(result, b.formatEndpoint(additional, "Additional: "))
+	}
+	return result
+}
+
+func (b *Builder) formatEndpoint(endpoint config.Endpoint, prefix string) string {
```
Comment (ajacquemot) [inline pkg/logs/status/builder.go]: Do we want to share this logic with the connection manager to log to which endpoint the agent connects to ?
see https://github.com/DataDog/datadog-agent/blob/master/pkg/logs/client/tcp/connection_manager.go#L60

**Review by ajacquemot — COMMENTED** [review]
LGTM, left a few nits, one enhancement and an open question.

File: pkg/status/dist/templates/logsagent.tmpl (hunk 3 above)
Comment (blemale) [reply pkg/status/dist/templates/logsagent.tmpl]: WDYT @NBParis ?

File: pkg/logs/status/builder.go (hunk 4 above)
Comment (blemale) [reply pkg/logs/status/builder.go]: I'm not convinced that we should share the logic as the connection manager works at another level, i.e the TCP level so it is not aware/doesn't care of the compression and the application layer, i.e HTTP.

File: pkg/status/dist/templates/logsagent.tmpl (hunk 3 above)
Comment (NBParis) [reply pkg/status/dist/templates/logsagent.tmpl]: Not sure about this, as this should be the first thing we show in the status, having the Log section starting with a descriptive sentence should be enough without requiring another header.

Comment (ajacquemot) [reply pkg/status/dist/templates/logsagent.tmpl]: sounds fair

File: pkg/logs/status/builder.go (hunk 4 above)
Comment (ajacquemot) [reply pkg/logs/status/builder.go]: sounds good

**Review by ajacquemot — DISMISSED** [review]

**Review by ajacquemot — APPROVED** [review]

**Pull request merged**